### ISSUE TITLE
teamsfx cli command: Configurable timeout looking up app registrations in Azure AD

### ISSUE THREAD

**Issue by tobim-dev** [OfficeDev/TeamsFx#4274]
**Is your feature request related to a problem? Please describe.**
We are using `teamsfx preview --local ` to start our teams app on our dev machines. When starting, `teamsfx` will check (and try to update) the corresponding app registration in the Azure Active Directory. However, our AD accounts do not have read permissions to see the app registration. `teamsfx preview` will start the app, but only after a timeout of about 30s. That increases our start up time for locally starting our teams app.  

**Describe the solution you'd like**
It would be great to provide the `teamsfx preview` command an additional flag to either not check for the app registration or to set the timeout to a lower value than 30s. 

**Describe alternatives you've considered**
Getting read access to the Azure Active Directory is not possible in our case. The app registrations are manually created by an admin. 

Thank you for answer!

**Comment by KennethBWSong** [comment]
Hi @tobim-dev , thanks for your feedback. In case that m365 account does not have permission to create/update Azure AD apps, you can follow the steps below to use an existing Azure AD app. Note that you will need to config the Azure AD app properly to successfully run your Teams app.
1. Open `.fx/config.{env name}.json`
2. Add "auth" under root, and fill in object id, client id, client secret and accessAsUserScopeId as below:
![image](https://user-images.githubusercontent.com/63089166/158755049-2cfe3b96-1407-4d48-90af-b49890635fa1.png)

And then TeamsFx will use the Azure AD app you provide. Note that we will still try to update the Azure AD app, but will not break if failed to update, and will give out notification instead.

Please have a try to see whether the following steps can help you. Thanks a lot!

**Comment by tobim-dev** [comment]
Hi @KennethBWSong, thank you for your quick reply. The solution that you mentioned is exactly what we are currently doing. 
However, as `teamsfx` is still trying to update the Azure AD app it take a long time for that before give out the notification. 
I would like to reduce that time period (or skip the Azure AD app update step completely), since it increases the time to start the app locally.

**Comment by KennethBWSong** [comment]
Hi @tobim-dev Thank you for your reply! Unfortunately we does not support totally skipping updating Azure AD app now. I will mark this as a feature request and let you know as soon as we have any update.

**Comment by wh-alice** [comment]
Close stale issues. Please create new issue then we can support.
